### ISSUE TITLE
Commit without username, email address - Repo cloned with token

### ISSUE THREAD

**Issue by madhubalakrishnan** [libgit2/libgit2sharp#1864]
Working on a tool that will pull source code from provided github repo (using github personal access token), allow users to edit source code, commit and push the source code into a branch of user choice. The users provide two inputs, 
1. GitHub Repo URL
2. GitHub Personal Access Token

Commit method requires message, username, email address. The tool does not collect the user's email address or name, but just the above two inputs.

Can a commit be done with GitHub Personal Access Token?

### Reproduction steps
Below code does NOT work when app is containerised or deployed in k8s, as no git config is available in the containers. BuildSignature returns NULL.
```
Signature gitSignature = RepoConfig.BuildSignature(DateTimeOffset.Now);
repo.Commit(CommitMessage, gitSignature, gitSignature);
```
Below code throws exception. Identity cannot be built using gitAccessToken
```
Signature gitSignature = new Signature(new Identity(gitAccessToken, gitAccessToken), DateTimeOffset.Now);
repo.Commit(CommitMessage, gitSignature, gitSignature);
```

### Expected behavior
Can a commit be completed using gitAccessToken?

### Version of LibGit2Sharp (release number or SHA1)
0.26.2
### Operating system(s) tested; .NET runtime tested
Mac, Windows, Linux - Containerised using Docker and deployed on K8s

**Comment by ethomson** [comment]
> Can a commit be done with GitHub Personal Access Token?

No.  A personal access token is how you authenticate to a service.  That has nothing to do with the structure of a commit, which requires an author and committer.  https://git-scm.com/book/en/v2/Git-Internals-Git-Objects

**Comment by madhubalakrishnan** [comment]
> No. A personal access token is how you authenticate to a service. That has nothing to do with the structure of a commit, which requires an author and committer. https://git-scm.com/book/en/v2/Git-Internals-Git-Objects

I agree, the difference of LibGit2Sharp CredentialsProvider vs. Committer/Author. When the tool does not collect the Username/Email Address, how can we commit the code? Surprisingly Java code/component used by a participating micro service is able to achieve this.

Any ideas? Suggestions?

**Comment by carlosmn** [comment]
If your tool does not want to ask the user for their name and email then you'll have to perform the commit as the tool or try to pick it up fro the current user's Git configuration (e.g. by using `Configuration.BuildSignature()` to try to get it from the configuration available to the user and repository). You definitely should *not* include an access token in a commit.

What exception do you get? It should allow you to put pretty much anything in there even if it's secret data.

**Comment by madhubalakrishnan** [comment]
@carlosmn : Thanks for your note.

We used BuildSignature() method, but when run on containers/K8s, this method returns a NULL. This is because the ~/.gitconfig file (git username, email cache file) is not present in a new container.

As you mentioned we did not try to use the token (secret) in a commit message/username, but we were trying to generate a signature based on the token.

As a work around we now use a standard username(ProductName) and emailaddress(product@company.com). 

I did compare how the Java component (`org.eclipse.jgit`) work without the gitconfig on containers. They do not return a NULL, but send a standard username as "root". Thus the code does not break when the .gitconfig file is not available. May be LibGit2Sharp method BuildSignature() could also adopt such an approach, rather than returning a NULL. NULL creates runtime error.
